src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/SplashScreen.cs

```diff
@@ -274,160 +275,156 @@ private void Fadeout_Tick(object unused, EventArgs args)
             else
             {
                 double progress = (_fadeoutEnd - dtNow).TotalMilliseconds / _fadeoutDuration.TotalMilliseconds;
-                _blendFunc.SourceConstantAlpha = (byte)(255 * progress);
-                unsafe
-                {
-                    UnsafeNativeMethods.UpdateLayeredWindow(_hwnd, IntPtr.Zero, null, null, IntPtr.Zero, null, 0, ref _blendFunc, NativeMethods.ULW_ALPHA);
-                }
+                _blendFunction.SourceConstantAlpha = (byte)(255 * progress);
+                PInvoke.UpdateLayeredWindow(_hwnd, HDC.Null, null, null, HDC.Null, null, default, _blendFunction, UPDATE_LAYERED_WINDOW_FLAGS.ULW_ALPHA);
             }
         }
 
-        private void DestroyResources()
+        private unsafe void DestroyResources(bool finalizer = false)
```

Comment:
Should there be an updated call somewhere where we are calling `DestroyResources(true)`? I don't see it here though I may have missed it.

Reply:
Yeah, good catch. I had added a finalizer so this wouldn't leak if misused and didn't move it over.